cmd/shared/shared.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package shared contains types and functions that are used across the full
+// set of mixer commands.
+package shared
+
+import (
+	"github.com/spf13/cobra"
+
+	"istio.io/mixer/pkg/version"
+)
+
+// OutFn is a function used for normal output.
+type OutFn func(format string, a ...interface{})
```

Comment:
Given this has been elevated to some shared code rather than being an implementation detail of a command, I wonder if this stuff would be better if we gave the commands an io.Writer instead of a function pointer? It would compose better and would still provide the same flexibility to capture all the output in tests...